tests/SpeechSynthesisTests.ts

```diff
@@ -612,4 +623,53 @@ describe.each([true, false])("Service based tests", (forceNodeWebSocket: boolean
             });
         });
     });
+
+    test("test Speech Synthesiser: Language Auto Detection", (done: jest.DoneCallback) => {
+        // tslint:disable-next-line:no-console
+        console.info("Name: test Speech Synthesiser, Language Auto Detection");
+        const speechConfig: sdk.SpeechConfig = BuildSpeechConfig();
+        objsToClose.push(speechConfig);
+        const autoDetectSourceLanguageConfig: sdk.AutoDetectSourceLanguageConfig = sdk.AutoDetectSourceLanguageConfig.fromOpenRange();
+        objsToClose.push(autoDetectSourceLanguageConfig);
+
+        const s: sdk.SpeechSynthesizer = sdk.SpeechSynthesizer.FromConfig(speechConfig, autoDetectSourceLanguageConfig, null);
+        objsToClose.push(s);
+        expect(s).not.toBeUndefined();
+
+        const con: sdk.Connection = sdk.Connection.fromSynthesizer(s);
+
+        con.messageSent = (args: sdk.ConnectionMessageEventArgs): void => {
+            if (args.message.path === "synthesis.context" && args.message.isTextMessage) {
+                try {
+                    expect(args.message.TextMessage).toContain(`\"autoDetection\":true`);
+                } catch (error) {
+                    done.fail(error);
+                }
+            }
+        };
+
+        s.SynthesisCanceled = (o: sdk.SpeechSynthesizer, e: sdk.SpeechSynthesisEventArgs): void => {
+            done.fail();
+        };
+
+        // we will get very short audio as the en-US voices are not mix-lingual.
+        s.speakTextAsync("你好世界。", (result: sdk.SpeechSynthesisResult): void => {
+            // tslint:disable-next-line:no-console
+            console.info("speaking finished, turn 1");
+            CheckSynthesisResult(result, sdk.ResultReason.SynthesizingAudioCompleted);
+            expect(result.audioData.byteLength).toBeGreaterThan(64 << 7); // longer than 1s
```

Comment:
General question about this test: is there something different about what is returned when openRange is set that can be checked?

Reply:
If `openRange` is set, the `synthesis.context` will contain `"autoDetection":true`. And, the synthesized text is Chinese, if no auto detection, the default english voices cannot read Chinese, then a empty result will be returned.